target/linux/mpc85xx/patches-4.9/102-powerpc-add-cmdline-override.patch

```diff
@@ -0,0 +1,37 @@
+--- a/arch/powerpc/Kconfig
++++ b/arch/powerpc/Kconfig
+@@ -707,6 +707,14 @@
+ 	  This is useful if you cannot or don't want to change the
+ 	  command-line options your boot loader passes to the kernel.
+
++config CMDLINE_OVERRIDE
++	bool "Use alternative cmdline from device tree"
++	help
++	  Some bootloaders may have uneditable bootargs. While CMDLINE_FORCE can
++	  be used, this is not a good option for kernels that are shared across
++	  devices. This setting enables using "chosen/cmdline-override" as the
++	  cmdline if it exists in the device tree.
++
+ config EXTRA_TARGETS
+ 	string "Additional default image types"
+ 	help
+--- a/drivers/of/fdt.c
++++ b/drivers/of/fdt.c
+@@ -1079,6 +1079,17 @@
+ 	if (p != NULL && l > 0)
+ 		strlcpy(data, p, min((int)l, COMMAND_LINE_SIZE));
+
++	/* CONFIG_CMDLINE_OVERRIDE is used to fallback to a different
++	 * device tree option of chosen/bootargs-override. This is
++	 * helpful on boards where u-boot sets bootargs, and is unable
++	 * to be modified.
++	 */
++#ifdef CONFIG_CMDLINE_OVERRIDE
++	p = of_get_flat_dt_prop(node, "bootargs-override", &l);
++	if (p != NULL && l > 0)
++		strlcpy(data, p, min((int)l, COMMAND_LINE_SIZE));
```

Comment:
If someone knows of a better way to do this, please let me know.

For some background, the Aerohive HiveAP 330 has an old, locked down version of u-boot that does not allow you to change it's bootargs, or it's bootcmd (which sets bootargs). Because of this, a workaround to set the cmdline is required.

I initially looked at using a cuImage to try and do this, but I had no luck as this board seems to have issues when I try to boot cuImages. I will admit my skills in that specific area are lacking, and this seemed like a cleaner, safer way to adjust the cmdline as no other dtb "fixups" are needed.

I also didn't want to force the kernel cmdline via the kernel config, due to the fact this board runs UART at 9600.